src/dag_algo/mod.rs

```diff
@@ -413,19 +419,28 @@ pub fn lexicographical_topological_sort(
     .into())
 }
 
-/// Return the topological generations of a DAG
+/// Return the topological generations of a directed graph.
 ///
-/// A topological generation is node collection in which ancestors of a node in each
-/// generation are guaranteed to be in a previous generation, and any descendants of
-/// a node are guaranteed to be in a following generation. Nodes are guaranteed to
-/// be in the earliest possible generation that they can belong to.
+/// A topological generation is a collection of nodes where all ancestors of a
+/// node are guaranteed to be in a previous generation, and all descendants of a
+/// node are guaranteed to be in a subsequent generation. Nodes are placed in
+/// the earliest possible generation they can belong to.
 ///
-/// :param PyDiGraph graph: The DAG to get the topological generations from
+///   >>> G = rx.PyDiGraph()
+///   >>> G.add_nodes_from([0, 1, 2, 3, 4])
+///   >>> G.add_edges_from_no_data([(0, 1), (0, 2), (1, 3), (2, 3), (3, 4)])
```

Comment:
ditto

Reply:
The same as below. Here even more: perfectly matching numbers only.